### PR TITLE
feat: add auth and testing to question-service

### PR DESCRIPTION
**Pull request by fullfatwasabi**

### CODE REVIEW

File: services/question-service/src/commons/auth/authenticator.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+// }
+
+// assertIsSelf
+// export async function assertIsSelfOrAdmin(req: Request, userId: number) {
+//   const token = await extractToken(req);
+//   if (!isSelfOrAdmin(token, userId)) {
+//     throw new HttpError("You do not have the correct permission", 401);
+//   }
+// }
+
```
Comment (aweijun) [inline services/question-service/src/commons/auth/authenticator.ts]: If this is not used jus remove?

File: services/question-service/src/tests/e2e.test.ts [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+const prisma = getPrismaClient();
+
+loadEnvConfig();
+
+jest.mock("next-auth/jwt");
+
+beforeEach(async () => {
+  await resetDb();
+  jest.clearAllMocks();
+  (getToken as jest.Mock).mockResolvedValue(null);
```
Comment (aweijun) [inline services/question-service/src/tests/e2e.test.ts:27]: What does this do?

**Review by aweijun — CHANGES_REQUESTED** [review]

File: services/question-service/src/tests/e2e.test.ts (hunk 2 above)
Comment (fullfatwasabi) [reply services/question-service/src/tests/e2e.test.ts:27]: just sets a default token result if no token provided ah

**Review by aweijun — APPROVED** [review]
LGTM

**Pull request merged**